### ISSUE TITLE
BinaryFocalCrossentropy: alpha does not work

### ISSUE THREAD

**Issue by yanghtr** [tensorflow/tensorflow#61519]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

tf 2.13

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 20.04

### Python version

3.10.12

### Current behavior?

`tf.keras.losses.BinaryFocalCrossentropy` computes the loss without using the `alpha`.
```
import tensorflow as tf
y_true_list = [0, 1, 0, 0]
logits_list = [-1.6, 0.51, 2.94, -1.8]
gamma = 2

focal_func1 = tf.keras.losses.BinaryFocalCrossentropy(gamma=gamma, alpha=0.25, from_logits=True)
focal_loss1 = focal_func1(y_true_list, logits_list)

focal_func2 = tf.keras.losses.BinaryFocalCrossentropy(gamma=gamma, alpha=10, from_logits=True)
focal_loss2 = focal_func2(y_true_list, logits_list)

focal_func3 = tf.keras.losses.BinaryFocalCrossentropy(gamma=gamma, alpha=100, from_logits=True)
focal_loss3 = focal_func3(y_true_list, logits_list)

print(focal_loss1)
print(focal_loss2)
print(focal_loss3)  
```
The results are:
```
tf.Tensor(0.6932789, shape=(), dtype=float32)
tf.Tensor(0.6932789, shape=(), dtype=float32)
tf.Tensor(0.6932789, shape=(), dtype=float32)
```

`focal_loss1` in the codes should be `0.51168`

### Standalone code to reproduce the issue

```shell
In above.
```

**Comment by vatsalkachhia** [comment]
You need to have "apply_class_balancing= True" if you want to see the effects of alpha.

Instead of :
tf.keras.losses.BinaryFocalCrossentropy(gamma=gamma, alpha=0.25, from_logits=True)

Try:
tf.keras.losses.BinaryFocalCrossentropy(gamma=gamma, alpha=0.25, from_logits=True, , apply_class_balancing= True)
